TESTS/integration/fs-threaded/main.cpp

```diff
@@ -102,39 +85,88 @@ static control_t file_2_threads(const size_t call_count)
     return CaseNext;
 }
 
-static control_t file_3_threads(const size_t call_count)
+static control_t file_4_threads(const size_t call_count)
 {
-    thread_counter = 0;
-    size_t block_size1 = 256;
-    size_t block_size2 = 1024;
-    size_t block_size3 = 4096;
+    size_t block_size1 = 4;
+    size_t block_size2 = 256;
+    size_t block_size3 = 1024;
+    size_t block_size4 = 4096;
 
     Thread t1;
     Thread t2;
     Thread t3;
+    Thread t4;
     t1.start(callback(file_fn, &block_size1));
     t2.start(callback(file_fn, &block_size2));
     t3.start(callback(file_fn, &block_size3));
+    t4.start(callback(file_fn, &block_size4));
     t1.join();
     t2.join();
     t3.join();
+    t4.join();
 
     return CaseNext;
 }
 
-utest::v1::status_t greentea_setup(const size_t number_of_cases)
+utest::v1::status_t test_setup_handler(const size_t number_of_cases)
 {
     GREENTEA_SETUP(10 * 60, "default_auto");
+
+    /* Format only once per each test run */
+    int format_err = fs.format(&sbd);
+    TEST_ASSERT_EQUAL_INT_MESSAGE(0, format_err, "could not format block device");
+
+    int mount_err = fs.mount(&sbd);
+    TEST_ASSERT_EQUAL_INT_MESSAGE(0, mount_err, "could not mount block device");
+
     return greentea_test_setup_handler(number_of_cases);
 }
 
+void test_teardown_handler(const size_t passed, const size_t failed, const failure_t failure)
+{
+    int mount_err = fs.unmount();
+    TEST_ASSERT_EQUAL_INT_MESSAGE(0, mount_err, "could not unmount block device");
+
+    greentea_test_teardown_handler(passed, failed, failure);
+}
+
+utest::v1::status_t case_setup_handler(const Case *const source, const size_t index_of_case)
+{
+    thread_counter = 0;
+
+    return greentea_case_setup_handler(source, index_of_case);
+}
+
+utest::v1::status_t case_teardown_handler(const Case *const source,
+                                          const size_t passed,
+                                          const size_t failed,
+                                          const failure_t reason)
+{
+    char filename[255] = { 0 };
+
+    for (uint32_t idx = 1; idx <= thread_counter; idx++) {
+        snprintf(filename, 255, "mbed-file-test-%" PRIu32 ".txt", idx);
```

Comment:
Can you make a constant for this to make sure that the formatting of the created and removed filenames doesn't drift out of sync?

Reply:
Ok, done.

Reply:
Thanks, changes look good to me then.